client/app/visualizations/counter/Renderer.jsx

```diff
@@ -0,0 +1,74 @@
+import { isFinite } from 'lodash';
+import React, { useState, useEffect } from 'react';
+import cx from 'classnames';
+import resizeObserver from '@/services/resizeObserver';
+import { RendererPropTypes } from '@/visualizations';
+
+import { getCounterData } from './utils';
+
+import './render.less';
+
+function getCounterStyles(scale) {
+  return {
+    OTransform: `scale(${scale})`,
+    MsTransform: `scale(${scale})`,
+    MozTransform: `scale(${scale})`,
+    WebkitTransform: `scale(${scale})`,
+    transform: `scale(${scale})`,
+  };
+}
+
+function getCounterScale(container) {
+  const inner = container.firstChild;
+  const scale = Math.min(container.offsetWidth / inner.offsetWidth, container.offsetHeight / inner.offsetHeight);
+  return Number(isFinite(scale) ? scale : 1).toFixed(2); // keep only two decimal places
+}
+
+export default function Renderer({ data, options, visualizationName }) {
+  const [scale, setScale] = useState('1.00');
+  const [container, setContainer] = useState(null);
+
+  useEffect(() => {
+    if (container) {
+      const unwatch = resizeObserver(container, () => {
+        setScale(getCounterScale(container));
+      });
+      return unwatch;
+    }
+  }, [container, setScale]);
```

Comment:
I'm probably not understanding how this works, but are you sure `setScale` is correct in the 2nd argument?

Reply:
Yes it's correct - when you use `useEffect`/`useMemo`/`useCallback` you should pass every dependency you'll use in a hook function because the function will capture them and then you'll need to create new function instance that will capture new dependencies values (you may omit some of dependencies or pass extra ones if you know what you do)

Reply:
It still feels to me like you indeed should pass dependencies that you use in the hook function, but only the ones that get changed over time. Also in [docs](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) they don't seem to include the setters in the dependency list

Reply:
@gabrieldutra You're right. It's also mentioned in hooks API reference. Thanks!

Reply:
Fixed